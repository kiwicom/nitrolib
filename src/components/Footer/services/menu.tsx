// eslint-disable-next-line import/prefer-default-export
export const getLinks = (lang: string) => [
  {
    id: 1,
    title: __("menu.terms_and_conditions"),
    url: `https://www.kiwi.com/${lang}/pages/content/legal/`,
  },
  {
    id: 2,
    title: __("menu.terms_of_use"),
    url: `https://www.kiwi.com/${lang}/pages/content/terms/`,
  },
  {
    id: 3,
    title: __("menu.privacy_policy"),
    url: `https://www.kiwi.com/${lang}/content/privacy/`,
  },
  {
    id: 4,
    title: __("menu.security"),
    url: `https://www.kiwi.com/${lang}/pages/security/`,
  },
];
