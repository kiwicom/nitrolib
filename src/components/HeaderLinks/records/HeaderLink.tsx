export type HeaderLink = {
  id: string,
  translation: string,
  newWindow: boolean,
  url: string,
};

export type SearchForm = {
  mode: string,
  destination: { type: string, name: string }, // TODO move proper search form into records @vacuum
  checkIn: Date,
  checkOut: Date | null,
  adults: number,
  children: number,
};

export type HeaderLinksContext =
  | "holidays"
  | "booking"
  | "kiwicom"
  | "pulkovoairport"
  | "kayakholidays";
