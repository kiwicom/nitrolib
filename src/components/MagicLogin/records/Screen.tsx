
export type Screen =
  | "intro"
  | "signUp"
  | "noAccount"
  | "kiwiLogin"
  | "facebookLogin"
  | "googleLogin"
  | "magicLink"
  | "signUpConfirmation"
  | "resetPassword"
  | "getSingleBooking";
