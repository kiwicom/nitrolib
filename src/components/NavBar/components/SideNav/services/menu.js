// @flow strict
import * as React from "react";
import * as R from "ramda";
import AccountCircle from "@kiwicom/orbit-components/lib/icons/AccountCircle";
import Deals from "@kiwicom/orbit-components/lib/icons/Deals";
import ContactEmail from "@kiwicom/orbit-components/lib/icons/ContactEmail";
import StarFull from "@kiwicom/orbit-components/lib/icons/StarFull";
import Trip from "@kiwicom/orbit-components/lib/icons/Trip";
import City from "@kiwicom/orbit-components/lib/icons/City";
import Kiwicom from "@kiwicom/orbit-components/lib/icons/Kiwicom";
import KiwicomCare from "@kiwicom/orbit-components/lib/icons/KiwicomCare";
import KiwicomGuarantee from "@kiwicom/orbit-components/lib/icons/KiwicomGuarantee";
import Code from "@kiwicom/orbit-components/lib/icons/Code";
import Suitcase from "@kiwicom/orbit-components/lib/icons/Suitcase";
import TermsAndConditions from "@kiwicom/orbit-components/lib/icons/TermsAndConditions";
import Settings from "@kiwicom/orbit-components/lib/icons/Settings";
import Security from "@kiwicom/orbit-components/lib/icons/Security";
import Facebook from "@kiwicom/orbit-components/lib/icons/Facebook";
import Instagram from "@kiwicom/orbit-components/lib/icons/Instagram";
import Twitter from "@kiwicom/orbit-components/lib/icons/Twitter";
import Linkedin from "@kiwicom/orbit-components/lib/icons/Linkedin";
import Chat from "@kiwicom/orbit-components/lib/icons/Chat";
import Exchange from "@kiwicom/orbit-components/lib/icons/Exchange";

import type { Brand } from "../../../../../records/Brand";

export const icons = {
  AccountCircle,
  Deals,
  ContactEmail,
  Facebook,
  Instagram,
  StarFull,
  Kiwicom,
  KiwicomCare,
  KiwicomGuarantee,
  Code,
  Suitcase,
  TermsAndConditions,
  Security,
  City,
  Twitter,
  Linkedin,
  Trip,
  Settings,
  Chat,
  Exchange,
};

const companyPagesUI = {
  stories: {
    title: __("content.stories"),
    Icon: icons.Trip,
  },
  security: {
    title: __("content.pages.security_title"),
    Icon: icons.Security,
  },
  about: {
    title: __("content.pages.about.title"),
    Icon: icons.City,
  },
  terms: {
    title: __("content.pages.legal.title"),
    Icon: icons.TermsAndConditions,
  },
  gdpr_terms: {
    title: __("content.pages.terms.title"),
    Icon: icons.TermsAndConditions,
  },
  guarantee: {
    title: __("content.pages.guarantee.title"),
    Icon: icons.KiwicomGuarantee,
  },
  privacy: {
    title: __("content.pages.privacy.title"),
    Icon: icons.Security,
  },
  careers: {
    title: __("content.pages.careers.title"),
    Icon: icons.Suitcase,
  },
  media: {
    title: __("content.pages.media_room"),
    Icon: icons.Kiwicom,
  },
  invite: {
    title: __("pages.buddy.landing.title"),
    Icon: icons.Deals,
  },
};

type PageItems = {
  [key: string]: {|
    title: string,
    Icon: React.ComponentType<{ className: string }>,
    link?: string,
  |},
};

export const getPagesItems = (brand: Brand): PageItems =>
  R.compose(
    R.mapObjIndexed((val, key) => R.merge(val, companyPagesUI[key])),
    R.filter(R.prop("enabled")),
    // $FlowExpected: some props are missing in certain objects
  )(brand.content.pages);

// TODO: missing translations common.instagram etc
const socialPagesUI = {
  instagram: {
    type: "instagram",
    Icon: icons.Instagram,
  },
  twitter: {
    type: "twitter",
    Icon: icons.Twitter,
    label: "common.twitter",
  },
  youtube: {
    type: "youtube",
    Icon: icons.Kiwicom,
  },
  linkedin: {
    type: "linkedin",
    Icon: icons.Linkedin,
  },
  blog: {
    type: "blog",
    Icon: icons.Kiwicom,
  },
  facebook: {
    type: "facebook",
    Icon: icons.Facebook,
    label: "common.facebook",
  },
};

type SocialMedia = {|
  type: string,
  enabled: boolean,
  Icon: React.ComponentType<{ className: string }>,
  link?: string,
  label?: string,
|};

export const getSocialMediaItems = (brand: Brand): SocialMedia[] =>
  R.compose(
    R.values,
    R.mapObjIndexed((val, key) => R.merge(val, socialPagesUI[key])),
    R.filter(R.prop("enabled")),
    // $FlowExpected: some props are missing in certain objects
  )(brand.content.media);
