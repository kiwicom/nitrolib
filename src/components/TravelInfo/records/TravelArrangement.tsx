
export type Sector = {
  from: string,
  to: string,
  stops: "0" | "1" | "2" | "3",
  note: string,
};

export type TravelArrangement = {
  takeOff: Sector,
  landing: Sector,
};
