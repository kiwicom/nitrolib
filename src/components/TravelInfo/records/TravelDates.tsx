
export type TravelDates = {
  from: string,
  to: string,
};
