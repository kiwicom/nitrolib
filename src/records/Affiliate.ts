export type Affiliate = {
  id: string,
  params: { [key: string]: string },
};
