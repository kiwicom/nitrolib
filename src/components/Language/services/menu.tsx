import { Language } from "../../../records/Languages";
import { language as languageConsts } from "../../../styles";

const getWidthConstant = (filteredLanguages: Language[]) =>
  filteredLanguages.length < languageConsts.limit ? 2 : 3; // 2 || 3 columns

export const getLanguageWrapperHeight = (filteredLanguages: Language[], flat: boolean) => {
  const widthConstant = flat ? 1 : getWidthConstant(filteredLanguages);

  // For filtered cases - enables to wrap to (2 || 3) cols every time
  const heightConstraint =
    Math.ceil(filteredLanguages.length / widthConstant) * languageConsts.row.height; // custom height depends on number of columns
  return filteredLanguages.length <= languageConsts.limit // && this.props.isMobile
    ? filteredLanguages.length * languageConsts.row.height // custom height depends on number of languages, only on mobile && when less languages than limit
    : heightConstraint;
};

export const getLanguageWrapperWidth = (filteredLanguages: Language[], flat: boolean) =>
  (flat ? 1 : getWidthConstant(filteredLanguages)) * languageConsts.row.width;
