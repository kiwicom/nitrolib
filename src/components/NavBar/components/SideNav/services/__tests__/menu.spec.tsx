 srict
import { brandDefault } from "../../../../../../records/Brand";
import { getPagesItems, getSocialMediaItems } from "../menu";

describe("#Sidenav service menu", () => {
  test("filters social media", () => {
    const socialmedia = getSocialMediaItems(brandDefault);

    expect(socialmedia).toMatchSnapshot();
  });

  test("fetches company section", () => {
    const company = getPagesItems(brandDefault);

    expect(company).toMatchSnapshot();
  });
});
