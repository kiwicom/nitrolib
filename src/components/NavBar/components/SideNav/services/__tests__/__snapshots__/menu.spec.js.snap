// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Sidenav service menu fetches company section 1`] = `
Object {
  "about": Object {
    "Icon": [Function],
    "enabled": true,
    "fully_qualified": false,
    "link": "/pages/content/about",
    "title": "content.pages.about.title",
  },
  "careers": Object {
    "Icon": [Function],
    "enabled": true,
    "fully_qualified": true,
    "link": "https://www.kiwi.com/jobs",
    "title": "content.pages.careers.title",
  },
  "faq": Object {
    "enabled": true,
    "fully_qualified": false,
    "link": "/helpcenter",
  },
  "gdpr_terms": Object {
    "Icon": [Function],
    "enabled": true,
    "fully_qualified": false,
    "link": "/pages/content/terms",
    "title": "content.pages.terms.title",
  },
  "guarantee": Object {
    "Icon": [Function],
    "enabled": true,
    "fully_qualified": false,
    "link": "/pages/guarantee",
    "title": "content.pages.guarantee.title",
  },
  "invite": Object {
    "Icon": [Function],
    "enabled": true,
    "link": "/pages/invite",
    "title": "pages.buddy.landing.title",
  },
  "media": Object {
    "Icon": [Function],
    "enabled": true,
    "fully_qualified": false,
    "link": "https://media.kiwi.com/",
    "title": "content.pages.media_room",
  },
  "privacy": Object {
    "Icon": [Function],
    "enabled": true,
    "fully_qualified": false,
    "link": "/content/privacy",
    "title": "content.pages.privacy.title",
  },
  "security": Object {
    "Icon": [Function],
    "enabled": true,
    "link": "/pages/security",
    "title": "content.pages.security_title",
  },
  "stories": Object {
    "Icon": [Function],
    "enabled": true,
    "link": "https://www.kiwi.com/stories/?utm_source=kiwicom&utm_medium=website",
    "title": "content.stories",
  },
  "terms": Object {
    "Icon": [Function],
    "enabled": true,
    "fully_qualified": false,
    "link": "/pages/content/legal",
    "title": "content.pages.legal.title",
  },
  "top_routes": Object {
    "enabled": true,
    "fully_qualified": false,
    "link": "/cheap-flights",
  },
}
`;

exports[`#Sidenav service menu filters social media 1`] = `
Array [
  Object {
    "Icon": [Function],
    "enabled": true,
    "link": "https://instagram.com/kiwicom247",
    "type": "instagram",
  },
  Object {
    "Icon": [Function],
    "enabled": true,
    "label": "common.twitter",
    "link": "https://twitter.com/kiwicom247",
    "type": "twitter",
  },
  Object {
    "Icon": [Function],
    "enabled": true,
    "link": "https://www.linkedin.com/company/Kiwi.com",
    "type": "linkedin",
  },
  Object {
    "Icon": [Function],
    "enabled": true,
    "label": "common.facebook",
    "link": "https://www.facebook.com/kiwicom247",
    "type": "facebook",
  },
]
`;
